lighthouse-cli/index.js

```diff
@@ -139,22 +145,73 @@ if (cli.verbose) {
   flags.logLevel = 'error';
 }
 
-// kick off a lighthouse run
-lighthouse(url, flags, config)
-  .then(results => Printer.write(results, outputMode, outputPath))
-  .then(results => {
-    if (outputMode !== 'html') {
-      Printer.write(results, 'html', './last-run-results.html');
-    }
-    return;
-  })
-  .catch(err => {
-    if (err.code === 'ECONNREFUSED') {
-      console.error('Unable to connect to Chrome. Please run Chrome w/ debugging port 9222 open:');
-      console.error('    npm explore -g lighthouse -- npm run chrome');
-    } else {
-      console.error('Runtime error encountered:', err);
-      console.error(err.stack);
+class Runner {
+  constructor() {
+    this.retries = 0;
+  }
+
+  launchChromeAndRetry() {
+    if (this.retries++ > 0) {
+      this.showConnectionError();
     }
+
+    const launcher = new Launcher({
+      head: !cli.selectChrome
+    });
+
+    return launcher
+      .run()
+      .then(() => {
+        console.log('Attempting to run again...');
+        return this.run();
+      })
+      .then(() => launcher.kill())
+      .catch(err => {
+        launcher.kill();
+        this.showRuntimeError(err);
```

Comment:
since `launchChromeAndRetry` is only called from `run()`, could this be 
```javascript
.catch(err => {
  launcher.kill();
  throw err;
});
```
and `run`'s `catch()` will deal with showing the runtime error?